src/main/java/jimmy/mcgymmy/logic/commands/EditCommand.java

```diff
@@ -11,84 +11,93 @@
             "John Doe",
             ParserUtil::parseName
     );
-    private OptionalParameter<Phone> phoneParameter = this.addOptionalParameter(
-            "phone",
+    private OptionalParameter<Protein> proteinParameter = this.addOptionalParameter(
+            "protein value",
             "p",
-            "Phone number of person to add",
-            "98765432",
-            ParserUtil::parsePhone
+            "Protein value of food (g)",
+            "10",
+            ParserUtil::parseProtein
     );
-    private OptionalParameter<Email> emailParameter = this.addOptionalParameter(
-            "email",
-            "e",
-            "Email address of person to add",
-            "johnd@example.com",
-            ParserUtil::parseEmail
+    private OptionalParameter<Fat> fatParameter = this.addOptionalParameter(
+            "fat value",
+            "f",
+            "Fat value of food (g)",
+            "10",
+            ParserUtil::parseFat
+    );
+    private OptionalParameter<Carbohydrate> carbParameter = this.addOptionalParameter(
+            "carb value",
+            "c",
+            "Carbohydrate value of food (g)",
+            "10",
+            ParserUtil::parseCarb
     );
 
     void setParameters(Parameter<Index> indexParameter, OptionalParameter<Name> nameParameter,
-                       OptionalParameter<Phone> phoneParameter, OptionalParameter<Email> emailParameter) {
+                       OptionalParameter<Protein> proteinParameter, OptionalParameter<Fat> fatParameter,
+                       OptionalParameter<Carbohydrate> carbParameter) {
         this.indexParameter = indexParameter;
         this.nameParameter = nameParameter;
-        this.phoneParameter = phoneParameter;
-        this.emailParameter = emailParameter;
+        this.proteinParameter = proteinParameter;
+        this.fatParameter = fatParameter;
+        this.carbParameter = carbParameter;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Food> lastShownList = model.getFilteredFoodList();
         Index index = indexParameter.consume();
 
         if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_FOOD_DISPLAYED_INDEX);
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Food foodToEdit = lastShownList.get(index.getZeroBased());
 
-        Name newName = this.nameParameter.getValue().orElseGet(personToEdit::getName);
-        Phone newPhone = this.phoneParameter.getValue().orElseGet(personToEdit::getPhone);
-        Email newEmail = this.emailParameter.getValue().orElseGet(personToEdit::getEmail);
+        Name newName = this.nameParameter.getValue().orElseGet(foodToEdit::getName);
```

Comment:
hmm we may be able to abstract this part a bit more in the future, just a note, no need to change now